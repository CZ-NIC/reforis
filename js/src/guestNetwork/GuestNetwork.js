/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import propTypes from 'prop-types';

import ForisForm from '../formContainer/ForisForm';
import GuestNetworkForm from './GuestNetworkForm';
import {APIEndpoints} from '../common/API';
import {validateIPv4Address} from '../common/validations';

GuestNetwork.propTypes = {
    ws: propTypes.object.isRequired
};

export default function GuestNetwork({ws}) {
    return <ForisForm
        ws={ws}
        forisConfig={{
            endpoint: APIEndpoints.guestNetwork,
            wsModule: 'guest'
        }}
        prepData={prepData}
        prepDataToSubmit={prepDataToSubmit}
        validator={validator}
    >
        <GuestNetworkForm/>
    </ForisForm>
}

function prepData(formData) {
    delete formData.interface_count;
    delete formData.interface_up_count;
    return formData
}

export function prepDataToSubmit(formData) {
    if (!formData.enabled)
        return {enabled: false};

    if (!formData.dhcp.enabled)
        formData.dhcp = {enabled: false};
    else
        delete formData.dhcp.clients;

    if (!formData.qos.enabled)
        formData.qos = {enabled: false};

    return formData
}

export function validator(formData) {
    const errors = {
        ip: validateIPv4Address(formData.ip),
        netmask: validateIPv4Address(formData.netmask),
    };
    return errors.ip || errors.netmask ? errors : undefined;
}
