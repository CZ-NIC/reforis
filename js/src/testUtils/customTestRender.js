/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from 'react';
import {UIDReset} from 'react-uid';
import {StaticRouter} from 'react-router';
import {render} from '@testing-library/react'

function Wrapper({children}) {
    return <StaticRouter>
        <UIDReset>
            {children}
        </UIDReset>
    </StaticRouter>
}

const customTestRender = (ui, options) => render(ui, {wrapper: Wrapper, ...options});

// re-export everything
export * from '@testing-library/react'

// override render method
export {customTestRender as render}
