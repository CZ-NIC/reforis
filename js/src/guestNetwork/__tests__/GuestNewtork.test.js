/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {fireEvent, getByLabelText, getByText, render, wait} from 'customTestRender';

import {mockedWS} from 'mockWS';
import mockAxios from 'jest-mock-axios';
import GuestNetwork from '../GuestNetwork';
import guestNetworkFixture from './__fixtures__/guestNetwork';


describe('<GuestNetwork/>', () => {
    let guestNetworkContainer;

    beforeEach(async () => {
        const mockWebSockets = new mockedWS();
        const {container} = render(<GuestNetwork ws={mockWebSockets} setFormValue={() => {}}/>);
        mockAxios.mockResponse({data: guestNetworkFixture()});
        await wait(() => getByLabelText(container, 'Enable'));
        guestNetworkContainer = container
    });

    it('Snapshot disabled.', () => {
        expect(guestNetworkContainer).toMatchSnapshot();
    });

    it('Snapshot enabled.', () => {
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable'));
        expect(guestNetworkContainer).toMatchSnapshot();
    });

    it('Snapshot enabled DHCP.', () => {
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable'));
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable DHCP'));
        expect(guestNetworkContainer).toMatchSnapshot();
    });

    it('Snapshot enabled QoS.', () => {
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable'));
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable QoS'));
        expect(guestNetworkContainer).toMatchSnapshot();
    });

    it('Test post.', async () => {
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable'));
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable DHCP'));
        fireEvent.click(getByLabelText(guestNetworkContainer, 'Enable QoS'));
        fireEvent.click(getByText(guestNetworkContainer, 'Save'));

        expect(mockAxios.post).toBeCalled();
        const data = {
            "dhcp": {
                "enabled": true,
                "lease_time": 3600,
                "limit": 150,
                "start": 100,
            },
            "enabled": true,
            "ip": "10.111.222.1",
            "netmask": "255.255.255.0",
            "qos": {
                "enabled": true,
                "download": 1023,
                "upload": 1025,
            },
        };
        expect(mockAxios.post).toHaveBeenCalledWith('/api/guest-network', data, expect.anything());
    });
});
