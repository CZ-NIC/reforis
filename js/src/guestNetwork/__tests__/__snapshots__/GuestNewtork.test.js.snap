// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestNetwork/> Snapshot disabled. 1`] = `
<div>
  <form>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="1"
        >
          Enable
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="1"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
    >
       
       
      Save
    </button>
  </form>
</div>
`;

exports[`<GuestNetwork/> Snapshot enabled DHCP. 1`] = `
<div>
  <form>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="16"
        >
          Enable
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="16"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="22"
      >
        Router IP address
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="22"
            required=""
            type="text"
            value="10.111.222.1"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          Router's IP address in the guest inner network.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="23"
      >
        Network mask
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="23"
            required=""
            type="text"
            value="255.255.255.0"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="24"
        >
          Enable DHCP
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="24"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          Enable this option to automatically assign IP addresses to the devices connected to the router.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="30"
      >
        DHCP start
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="30"
            min="1"
            required=""
            type="number"
            value="100"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="31"
      >
        DHCP max leases
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="31"
            min="1"
            required=""
            type="number"
            value="150"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="32"
      >
        Lease time (hours)
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="32"
            min="120"
            required=""
            type="number"
            value="3600"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="25"
        >
          Enable QoS
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="25"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          This option enables you to set a bandwidth limit for the guest network, so that your main network doesn't get slowed-down by it.
        </small>
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
    >
       
       
      Save
    </button>
  </form>
</div>
`;

exports[`<GuestNetwork/> Snapshot enabled QoS. 1`] = `
<div>
  <form>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="34"
        >
          Enable
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="34"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="40"
      >
        Router IP address
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="40"
            required=""
            type="text"
            value="10.111.222.1"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          Router's IP address in the guest inner network.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="41"
      >
        Network mask
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="41"
            required=""
            type="text"
            value="255.255.255.0"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="42"
        >
          Enable DHCP
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="42"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          Enable this option to automatically assign IP addresses to the devices connected to the router.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="43"
        >
          Enable QoS
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="43"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          This option enables you to set a bandwidth limit for the guest network, so that your main network doesn't get slowed-down by it.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="49"
      >
        Download
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="49"
            min="1"
            required=""
            type="number"
            value="1023"
          />
          <div
            class="input-group-append"
          >
            <p
              class="input-group-text"
            >
              kb/s
            </p>
          </div>
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          Download speed in guest network (in kilobits per second).
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="50"
      >
        Upload
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="50"
            min="1"
            required=""
            type="number"
            value="1025"
          />
          <div
            class="input-group-append"
          >
            <p
              class="input-group-text"
            >
              kb/s
            </p>
          </div>
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          Upload speed in guest network (in kilobits per second).
        </small>
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
    >
       
       
      Save
    </button>
  </form>
</div>
`;

exports[`<GuestNetwork/> Snapshot enabled. 1`] = `
<div>
  <form>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="6"
        >
          Enable
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="6"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="12"
      >
        Router IP address
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="12"
            required=""
            type="text"
            value="10.111.222.1"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          Router's IP address in the guest inner network.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <label
        class="form-control-label col-sm-3"
        for="13"
      >
        Network mask
      </label>
      <div
        class="col-sm-8"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control "
            id="13"
            required=""
            type="text"
            value="255.255.255.0"
          />
        </div>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        />
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="14"
        >
          Enable DHCP
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="14"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          Enable this option to automatically assign IP addresses to the devices connected to the router.
        </small>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="form-label col-sm-3"
      >
        <label
          class="form-label"
          for="15"
        >
          Enable QoS
        </label>
      </div>
      <div
        class="form-check col-sm-8"
      >
        <input
          id="15"
          type="checkbox"
        />
        <small
          class="form-text text-muted"
        >
          This option enables you to set a bandwidth limit for the guest network, so that your main network doesn't get slowed-down by it.
        </small>
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
    >
       
       
      Save
    </button>
  </form>
</div>
`;
